src/instructlab/model/train.py

```diff
@@ -496,219 +498,267 @@ def get_files(directory: str, pattern: str) -> list[str]:
+                # Right now it's just copying files from the first one numerically not necessarily the best one
+                for fpath in (
+                    "checkpoint-*/added_tokens.json",
+                    "checkpoint-*/special_tokens_map.json",
+                    "checkpoint-*/tokenizer.json",
+                    "checkpoint-*/tokenizer.model",
+                    "checkpoint-*/tokenizer_config.json",
+                    "merged_model/config.json",
+                    "merged_model/generation_config.json",
+                ):
+                    file_ = next(training_results_dir.glob(fpath))
+                    shutil.copy(file_, final_results_dir)
+                    print(f"Copied {file_} to {final_results_dir}")
+
+                for file in training_results_dir.glob("merged_model/*.safetensors"):
+                    shutil.move(file, final_results_dir)
+                    print(f"Moved {file} to {final_results_dir}")
+
+                if four_bit_quant:
+                    print(
+                        "SKIPPING CONVERSION to gguf. This is unsupported with --4-bit-quant. "
+                        + "See https://github.com/instructlab/instructlab/issues/579."
+                    )
+                    return
+
+                gguf_file_path = convert_llama_to_gguf(
+                    model=final_results_dir, pad_vocab=True
                 )
 
-            _prepare_phased_base_dir(phased_base_dir)
-
-            _run_phased_training(
-                ctx=ctx,
-                train_args=train_args,
-                torch_args=torch_args,
-                base_dir=phased_base_dir,
-                phase1_data=phased_phase1_data,
-                phase1_num_epochs=phased_phase1_num_epochs,
-                phase1_samples_per_save=phased_phase1_samples_per_save,
-                phase1_checkpoints_dir=phased_base_dir / "phase1" / "checkpoints",
-                phased_phase1_effective_batch_size=phased_phase1_effective_batch_size,
-                phase2_data=phased_phase2_data,
-                phase2_num_epochs=phased_phase2_num_epochs,
-                phase2_samples_per_save=phased_phase2_samples_per_save,
-                phase2_checkpoints_dir=phased_base_dir / "phase2" / "checkpoints",
-                phased_phase2_effective_batch_size=phased_phase2_effective_batch_size,
-                phase2_eval_cache=phased_base_dir / "phase2" / "eval_cache",
-                mtbench_judge=mt_bench_judge,
-                enable_serving_output=enable_serving_output,
-            )
+                # Remove safetensors files to save space, were done with them here
+                # and the huggingface lib has them cached
+                for file in final_results_dir.glob("*.safetensors"):
+                    file.unlink()
+
+                shutil.move(gguf_file_path, gguf_models_file)
+                print(f"Save trained model to {gguf_models_file}")
```

Comment:
Use logger here instead of print

Reply:
Was this fixed?

Reply:
fixed, for the record this is the existing linux train code. I'd like to not change --pipeline simple too much and instead opt for those enhancements to be in a different PR when we re-write MLX and Linux simple train

Reply:
getting logging fixes in now will reduce the size/review burden on a future PR